moto/servicecatalog/models.py

```diff
@@ -295,5 +371,97 @@ def describe_portfolio(
 
         return portfolio_detail, tags, tag_options, budgets
 
+    def create_product(
+        self,
+        name: str,
+        owner: str,
+        description: Optional[str],
+        distributor: Optional[str],
+        support_description: Optional[str],
+        support_email: Optional[str],
+        support_url: Optional[str],
+        product_type: str,
+        tags: list[dict[str, str]],
+        provisioning_artifact_parameters: Optional[dict[str, Any]],
+        idempotency_token: Optional[str] = None,
+        source_connection: Optional[dict[str, Any]] = None,
+        accept_language: Optional[str] = None,
+    ) -> Product:
+        if idempotency_token and idempotency_token in self.idempotency_tokens:
+            product_id = self.idempotency_tokens[idempotency_token]
+            product = self.products[product_id]
+            return product
+
+        token = idempotency_token or str(uuid.uuid4())
+        existing_id = self.idempotency_tokens.get(token)
+        if existing_id:
+            return self.products[existing_id]
```

Comment:
Doesn't this do the same as the if-statement above? Or am I misunderstanding the purpose of it?